include/benchmark/benchmark.h

```diff
@@ -1418,6 +1424,25 @@ class BENCHMARK_DEPRECATED_MSG("The CSV Reporter will be removed in a future rel
   std::set<std::string> user_counter_names_;
 };
 
+// If a MemoryManager is registered, it can be used to report memory statistics
+// for a run of the benchmark.
+class MemoryManager {
+ public:
+  struct Result {
+    explicit Result(int64_t iters)
+        : num_allocs(0), max_bytes_used(0), iterations(iters) {}
+    int64_t num_allocs;
+    int64_t max_bytes_used;
```

Comment:
It would be great to have a comment explaining _what_ this is.
Is this the total amount of memory allocated, ever, or the peak memory usage?

For reference, it is possible to get the former natively - e.g. [`llvm::sys::Process::GetMallocUsage()`](https://github.com/llvm-mirror/llvm/blob/09f0f7ccc5eec100c3902c070f674a5c97a17c92/lib/Support/Unix/Process.inc#L87-L114)

Reply:
I imagine we might have a native implementation of these things, but we might have someone with a custom allocator, or an integration with tcmalloc. I wanted to get the interface in so we can do something, and then worry about a native implementation (and all the cross-platform shenanigans that entails).